gapis/api/templates/api.go.tmpl

```diff
@@ -340,18 +485,65 @@ import (
     return s
   }
 
+  func (m {{$name}}) Keysʷ(ϟctx context.Context, ϟw ϟapi.StateWatcher) []{{$key}} {
+    if m.Map == nil {
+      return nil
+    }
+    if ϟw != nil {
+      ϟw.OnGet(ϟctx, m, ϟapi.CompleteFragment{}, ϟapi.NilReference{})
+    }
+    s := make({{$sorted}}, len(*m.Map))
+    i := 0
+    for k, _ := range *m.Map {
+      s[i] = k
+      i++
+    }
+    sort.Sort(s)
+    return s
+  }
+
   // Clone makes a deep copy of this map.
   func (m {{$name}}) Clone(ϟa arena.Arena, ϟseen ϟapi.CloneContext) {{$name}} {
-    if existing, ok := ϟseen[m.Map]; ok {
-      return {{$name}}{Map: existing.(*map[{{$key}}]{{$value}}), a: ϟa}
+    if existing, ok := ϟseen[m.ϟrefID]; ok {
+      return existing.({{$name}})
     }
     out := make(map[{{$key}}]{{$value}}, len(*m.Map))
-    ϟseen[m.Map] = &out
```

Comment:
This was here on purpose - before the recursion - to handle cycles.

Reply:
I have changed all the `ϟseen` values like this to be set in `Clone` before any recursion.